### PR TITLE
Implemented endpoint to fetch popular channels

### PR DESCRIPTION
**Pull request by devos50**
They are displayed on the front page in the new GUI.

### CODE REVIEW

**Comment by devos50** [comment]
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 825, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "<string>", line 2, in _do_request
    
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/RemoteTorrentHandler.py", line 32, in pass_when_stopped
    return f(self, *argv, **kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/RemoteTorrentHandler.py", line 455, in _do_request
    trackers = self._torrent_db_handler.getTrackerListByInfohash(infohash)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 696, in getTrackerListByInfohash
    torrent_id = self.getTorrentID(infohash)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 242, in getTorrentID
    return self.getTorrentIDS([infohash, ]).get(infohash)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 262, in getTorrentIDS
    torrents = self._db.fetchall(sql_stmt, to_select)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/dispersy/util.py", line 46, in helper
    return blockingCallFromThread(reactor, func, *args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/dispersy/util.py", line 225, in blockingCallFromThread
    return f(*args, **kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 401, in fetchall
    res = self.execute_read(sql, args)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 310, in execute_read
    return self.execute(sql, args)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/dispersy/util.py", line 46, in helper
    return blockingCallFromThread(reactor, func, *args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/dispersy/util.py", line 225, in blockingCallFromThread
    return f(*args, **kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 263, in execute
    cur = self.get_cursor()
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 196, in get_cursor
    self._cursor_table[thread_name] = self._connection.cursor()
exceptions.AttributeError: 'NoneType' object has no attribute 'cursor'
```

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Tests on mac timed out (already created an issue for it)

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@lfdversluis can you review this one? I will add an error when the limit <= 0.

File: Tribler/Core/Modules/restapi/channels/channels_popular_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                        "torrents": 3,
+                        "spam": 5,
+                        "modified": 14598395,
+                    }]
+                }
+        """
+        limit_channels = 10
+
+        if 'limit' in request.args and len(request.args['limit']) > 0:
+            limit_channels = request.args['limit'][0]
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/channels/channels_popular_endpoint.py]: Do not forget to error here on <= 0 as described in the other PR.

**Comment by lfdversluis** [comment]
Besides my comment, looks good.

**Comment by devos50** [comment]
@whirm can you review?

**Pull request merged**